provider/views.py

```diff
@@ -261,12 +263,22 @@ def handle(self, request, post_data=None):
 
         authorization_form = self.get_authorization_form(request, client,
             post_data, data)
-
-        if not authorization_form.is_bound or not authorization_form.is_valid():
-            return self.render_to_response({
-                'client': client,
-                'form': authorization_form,
-                'oauth_data': data, })
+        
+        if data.get('response_type', None):
+            if not authorization_form.is_bound \
+                or not authorization_form.is_valid():
+                return self.render_to_response({
+                                                'client': client,
+                                                'form': authorization_form,
+                                                'oauth_data': data, })
+
+            if data.get('response_type') == 'token':
+                #uses request.user as the urls.py already required login_required
+                atm = AccessTokenModel.objects.create(user=request.user,
```

Comment:
the base provider views check a `SINGLE_ACCESS_TOKEN` constant and decide whether or not to issue a new token or reuse an existing one (if exists).  Would be ideal to maintain consistency with that behavior here.